flink-runtime/src/main/java/org/apache/flink/runtime/state/StateBackendLoader.java

```diff
@@ -103,6 +108,14 @@ public static StateBackend loadStateBackendFromConfig(
         String factoryClassName = backendName;
 
         switch (backendName.toLowerCase()) {
+            case HASHMAP_STATE_BACKEND_NAME:
+                HashMapStateBackend hashMapStateBackend =
+                        new HashMapStateBackendFactory().createFromConfig(config, classLoader);
+                if (logger != null) {
+                    logger.info("State backend is set to heap memory {}", hashMapStateBackend);
+                }
+                return hashMapStateBackend;
+
             case MEMORY_STATE_BACKEND_NAME:
                 MemoryStateBackend memBackend =
                         new MemoryStateBackendFactory().createFromConfig(config, classLoader);
```

Comment:
Why do we still need to create and return deprecated `MemoryStateBackend`? Why can't we return `HashMapStateBackend` as above?

ditto `FS_STATE_BACKEND_NAME`